### PR TITLE
ocamlPackages.opti: init at 1.0.3

### PR DESCRIPTION
**Pull request by magnusjonsson**
###### Motivation for this change

Help promote nix at work by making this package available.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions (Arch Linux)
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by magnusjonsson** [comment]
Hi, this is my first pull request to this project so please bear with me :). Happy to make any changes or additional testing if needed. Thanks!

**Comment by vbgl** [comment]
@GrahamcOfBorg build ocamlPackages.opti

File: pkgs/development/ocaml-modules/opti/default.nix [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{ stdenv, fetchurl, buildDunePackage }:
+
+buildDunePackage rec {
+  pname = "opti";
+  version = "1.0.3";
+
+  src = fetchurl {
```
Comment (kalbasit) [inline pkgs/development/ocaml-modules/opti/default.nix]: please use `fetchFromGitHub` instead of fetchurl

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/development/ocaml-modules/opti/default.nix [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    url = "https://github.com/magnusjonsson/opti/releases/download/${version}/opti-${version}.tbz";
+    sha256 = "ed9ba56dc06e9d2b1bf097964cc65ea37db787d4f239c13d0dd74693f5b50a1e";
+  };
+
+  meta = {
```
Comment (kalbasit) [inline pkgs/development/ocaml-modules/opti/default.nix]: if you would write this as `meta = with stdenv.lib; {` then you can remove the `stdenv.lib` from line 14 and 15.

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/development/ocaml-modules/opti/default.nix (hunk 1 above)
Comment (catern) [reply pkgs/development/ocaml-modules/opti/default.nix]: No, fetchurl is valid here. This is fetching a Github release tarball. fetchFromGitHub is for fetching the autogenerated tarball for a Git tag, it doesn't support fetching a release tarball.

Comment (kalbasit) [reply pkgs/development/ocaml-modules/opti/default.nix]: I've checked the difference between both sources and I agree with @catern here as the released tarball has the version defined in it.

```path
diff -Naur not-generated/.gitignore generated/.gitignore
--- not-generated/.gitignore    1969-12-31 16:00:00.000000000 -0800
+++ generated/.gitignore        2018-11-30 11:35:52.149183398 -0800
@@ -0,0 +1,27 @@
+*.annot
+*.cmo
+*.cma
+*.cmi
+*.a
+*.o
+*.cmx
+*.cmxs
+*.cmxa
+
+# ocamlbuild working directory
+_build/
+
+# ocamlbuild targets
+*.byte
+*.native
+
+# oasis generated files
+setup.data
+setup.log
+
+* emacs generated backups
+*~
+
+# dune generated files
+*.merlin
+*.install
diff -Naur not-generated/opti.opam generated/opti.opam
--- not-generated/opti.opam     2018-11-25 00:08:19.000000000 -0800
+++ generated/opti.opam 2018-11-30 11:35:52.151183390 -0800
@@ -1,4 +1,3 @@
-version: "1.0.3"
 opam-version: "2.0"
 synopsis: "DSL to generate fast incremental C code from declarative specifications"
 description: "Generate C code to incrementally compute sums, products and other expressions, without recomputing them from scratch every time."
@@ -16,4 +15,4 @@
 build: [
   ["dune" "build" "-p" "opti" "-j" jobs]
   ["dune" "runtest" "-p" "opti" "-j" jobs] {with-test}
-]
\ No newline at end of file
+]
```

**Comment by magnusjonsson** [comment]
I believe this is ready to be merged.

**Review by kalbasit — APPROVED** [review]

**Pull request merged**